components/eamxx/src/share/util/eamxx_time_interpolation.cpp

```diff
@@ -0,0 +1,340 @@
+#include "share/util/eamxx_time_interpolation.hpp"
+
+namespace scream{
+namespace util {
+
+/*-----------------------------------------------------------------------------------------------*/
+// Constructors
+TimeInterpolation::TimeInterpolation(
+  const grid_ptr_type& grid 
+)
+{
+  // Given the grid initialize field managers to store interpolation data
+  m_fm_time0 = std::make_shared<FieldManager>(grid);
+  m_fm_time1 = std::make_shared<FieldManager>(grid);
+  m_fm_time0->registration_begins();
+  m_fm_time0->registration_ends();
+  m_fm_time1->registration_begins();
+  m_fm_time1->registration_ends();
+}
+/*-----------------------------------------------------------------------------------------------*/
+TimeInterpolation::TimeInterpolation(
+  const grid_ptr_type& grid, 
+  const vos_type& list_of_files
+) : TimeInterpolation(grid)
+{
+  set_file_data_triplets(list_of_files);
+}
+/*-----------------------------------------------------------------------------------------------*/
+/* A function to perform time interpolation using data from all the fields stored in the local
+ * field managers.
+ * Conducts a simple linear interpolation between two points using
+ *        y* = w*y0 + (1-w)*y1
+ * where w = (t1-t*)/(t1-t0)
+ * Input:
+ *   time_in - A timestamp to interpolate onto.
+ * Output
+ *   A map of (field name) and (Field) pairs such that each field is the interpolated values and
+ *   the map makes it possible to quickly query individual outputs.
+ */
+void TimeInterpolation::perform_time_interpolation(const TimeStamp& time_in)
+{
+  // Declare the output map
+  std::map<std::string,Field> interpolated_fields;
+
+  // If data is handled by files we need to check that the timestamps are still relevant
+  if (m_file_data_triplets.size()>0) {
+    check_and_update_data(time_in);
+  }
+
+  // Gather weights for interpolation.  Note, timestamp differences are integers and we need a
+  // real defined weight.
+  const Real w_num = m_time1 - time_in;
+  const Real w_den = m_time1 - m_time0;
+  const Real weight = w_num/w_den;
+
+  // Cycle through all stored fields and conduct the time interpolation
+  for (auto name : m_field_names)
+  {
+    const auto& field0   = m_fm_time0->get_field(name);
+    const auto& field1   = m_fm_time1->get_field(name);
+          auto field_out = m_interp_fields.at(name);
+    field_out.deep_copy(field0);
+    field_out.update(field1,1.0-weight,weight);
+  }
+}
+/*-----------------------------------------------------------------------------------------------*/
+/* Function which registers a field in the local field managers.
+ * Input:
+ *   field_in - Is a field with the appropriate dimensions and metadata to match the interpolation
+ * Output:
+ *   None
+ */
+void TimeInterpolation::add_field(Field& field_in, const bool deep)
```

Comment:
`deep` is not a very clear name, imho. At first, I actually thought that if `deep=true`, you should create a deep copy of the field, while it's used the other way around in the code.

Perhaps you could call the arg something like `store_soft_copy`.

Reply:
Also, you can take a const ref to `field_in`, since you are not going to change it.